code/game/machinery/computer/dna_console.dm

```diff
@@ -147,28 +205,56 @@
 	else
-		buttons += "<span class='linkOff'>Inject Rejuvenators</span>"
+		buttons += "<span class='linkOff'>Scramble DNA</span>"
 	if(diskette)
-		buttons += "<a href='?src=[REF(src)];task=ejectdisk'>Eject Disk</a>"
+		buttons += "<a href='?src=[REF(src)];task=screen;text=disk;'>Disk</a>"
 	else
-		buttons += "<span class='linkOff'>Eject Disk</span>"
-	if(current_screen == "buffer")
-		buttons += "<a href='?src=[REF(src)];task=screen;text=mainmenu;'>Radiation Emitter Menu</a>"
+		buttons += "<span class='linkOff'>Disk</span>"
+	if(current_screen == "mutations")
+		buttons += "<span class='linkOff'>Mutations</span>"
 	else
-		buttons += "<a href='?src=[REF(src)];task=screen;text=buffer;'>Buffer Menu</a>"
-
+		buttons += "<a href='?src=[REF(src)];task=screen;text=mutations;'>Mutations</a>"
+	if((current_screen == "mainmenu") || !current_screen)
+		buttons += "<span class='linkOff'>Genetic Sequencer</span>"
+	else
+		buttons += "<a href='?src=[REF(src)];task=screen;text=mainmenu;'>Genetic Sequencer</a>"
+	if(current_screen == "ui")
+		buttons += "<span class='linkOff'>Unique Identifiers</span>"
+	else
+		buttons += "<a href='?src=[REF(src)];task=screen;text=ui;'>Unique Identifiers</a>"
+	if(current_screen == "advinjector")
+		buttons += "<span class='linkOff'>Adv. Injectors</span>"
+	else
+		buttons += "<a href='?src=[REF(src)];task=screen;text=advinjector;'>Adv. Injectors</a>"
 	switch(current_screen)
 		if("working")
 			temp_html += status
 			temp_html += "<h1>System Busy</h1>"
 			temp_html += "Working ... Please wait ([DisplayTimeText(radduration*10)])"
-		if("buffer")
+		if("ui")
 			temp_html += status
 			temp_html += buttons
-			temp_html += "<h1>Buffer Menu</h1>"
+			temp_html += "<h1>Unique Identifiers</h1>"
+			temp_html += "<a href='?src=[REF(src)];task=setstrength;num=[radstrength-1];'>--</a> <a href='?src=[REF(src)];task=setstrength;'>Output Level</a> <a href='?src=[REF(src)];task=setstrength;num=[radstrength+1];'>++</a>"
+			temp_html += "<br><a href='?src=[REF(src)];task=setduration;num=[radduration-1];'>--</a> <a href='?src=[REF(src)];task=setduration;'>Pulse Duration</a> <a href='?src=[REF(src)];task=setduration;num=[radduration+1];'>++</a>"
+			temp_html += "<h3>Irradiate Subject</h3>"
+			temp_html += "<div class='line'><div class='statusLabel'>Unique Identifier:</div><div class='statusValue'><div class='clearBoth'>"
+			var/max_line_len = 7*DNA_BLOCK_SIZE
+			if(viable_occupant)
+				temp_html += "<div class='dnaBlockNumber'>1</div>"
+				var/len = length(viable_occupant.dna.uni_identity)
+				for(var/i=1, i<=len, i++)
+					temp_html += "<a class='dnaBlock' href='?src=[REF(src)];task=pulseui;num=[i];'>[copytext(viable_occupant.dna.uni_identity,i,i+1)]</a>"
+					if ((i % max_line_len) == 0)
+						temp_html += "</div><div class='clearBoth'>"
+					if((i % DNA_BLOCK_SIZE) == 0 && i < len)
+						temp_html += "<div class='dnaBlockNumber'>[(i / DNA_BLOCK_SIZE) + 1]</div>"
```

Comment:
Remember to readd unicode support changes here.

Reply:
again
wut